sympy/codegen/ast.py

```diff
@@ -452,15 +465,18 @@ def body(self):
 class Token(Basic):
     """ Similar to Symbol, but takes no assumptions.
 
-    Defining fields are set in __slots__.
+    Defining fields are set in __slots__. Attributes (defined in __slots__) are
+    allowed to contain instances of Basic or strings.
     """
 
     __slots__ = []
 
     def __new__(cls, *args, **kwargs):
         if len(args) == 1 and not kwargs and isinstance(args[0], cls):
             return args[0]
-        args = args + tuple([kwargs[k] for k in cls.__slots__[len(args):]])
+        args = args + tuple([kwargs.pop(k) for k in cls.__slots__[len(args):]])
+        if kwargs:
+            raise ValueError("Unknown kwargs: %s" % kwargs)
         obj = Basic.__new__(cls)
```

Comment:
ok, it's usually `str` or `None` that causes problems, (e.g. `Variable.type` can now be `None`), representing type through attributes would change that though.